manual/development.md

```diff
@@ -16,3 +16,189 @@ Do 3 steps:
   2. Add an entry into config/enabled.yml or config/disabled.yml
   3. Implement your new cop in the generated file!
 ```
+
+## Implementing the cop
+
+You can start by learning the
+[node pattern](http://www.rubydoc.info/gems/rubocop/RuboCop/NodePattern)
+and using the pattern to match with specific nodes that you want to match.
+
+If you're not familiar with node pattern, try to investigate how it goes by
+creating a few examples with code that you want to match. Let's start with a
+simple cop to simplify conditions with arrays:
+
+### Inspecting the AST representation
+
+Let's imagine we want to simplify statements from `!array.empty?` to
+`array.any?`:
+
+First, check what the bad code returns in the Abstract Syntax Tree
+representation. Let's debug our expression using the REPL from RuboCop:
+
+```sh
+$ rake repl
+```
+
+Now, we need to create an AST representation to match with it:
+
+
+```ruby
```

Comment:
Btw, you can do pretty much the same things with the parser API directly. That's what I usually do, although probably using RuboCop's API is a bit better idea in general.

Reply:
Indeed. I use the `ruby-parse` command line utility on a daily basis. 🙂

Reply:
I initially referred to `ruby-parse` because I also used it 😄
I was just confused if I should put it or not. I think maybe it worth to mention `ruby-parse` since looks like everybody uses it to check the AST representation.

Reply:
As far as I remember, the major difference is that `Parser::AST::Node` does not have `parent` attribute while RuboCop does. But for checking the representation makes sense stick with a simple `ruby-parse`.

Reply:
Exactly.